# HG changeset patch
# User Rocco Rutte <pdmef@gmx.net>
# Date 1233066304 -3600
# Branch HEAD
# Node ID eb918af802ec62136f5a522dcb1ac18cb470d4ac
# Parent  da94a92c3ba0ec7718c78a1c60d62931d13b062f
Unconditionally inode-sort Maildir and MH folders. Closes #3166.

diff -r da94a92c3ba0 -r eb918af802ec mh.c
--- a/mh.c	Mon Jan 26 12:32:30 2009 -0800
+++ b/mh.c	Tue Jan 27 15:25:04 2009 +0100
@@ -1027,13 +1027,14 @@
     if (!ctx->quiet && progress)
       mutt_progress_update (progress, count, -1);
 
+    DO_SORT();
+
     snprintf (fn, sizeof (fn), "%s/%s", ctx->path, p->h->path);
 
 #if USE_HCACHE
     if (option(OPTHCACHEVERIFY))
     {
-      DO_SORT();
-      ret = stat(fn, &lastchanged);
+       ret = stat(fn, &lastchanged);
     }
     else
     {
@@ -1057,7 +1058,6 @@
     {
 #endif /* USE_HCACHE */
 
-    DO_SORT();
     if (maildir_parse_message (ctx->magic, fn, p->h->old, p->h))
     {
       p->header_parsed = 1;
